Enhancer/PayloadEnhancer.php

```diff
@@ -24,9 +24,13 @@ class PayloadEnhancer implements EnhancerInterface
     /**
      * {@inheritDoc}
      */
-    public function enhance(Context $context, Resource $resource)
+    public function enhance(array $data, Context $context, Resource $resource)
     {
         $visitor = $context->getVisitor();
-        $visitor->addData('payload', $context->accept($resource->getPayload()));
+        $payload = $resource->getPayload();
+
+        $data['payload'] = $payload;
```

Comment:
Damn, the solution here was far too simple. I think this means that we can remove the `Context` dependency from the Enhancers.